File: stasis-core/src/de/boxxit/stasis/MessageException.java
package de.boxxit.stasis;

import com.esotericsoftware.kryo.DefaultSerializer;
import de.boxxit.stasis.serializer.MessageExceptionSerializer;

/**
 * User: Christian Fruth
 */
@DefaultSerializer(MessageExceptionSerializer.class)
public class MessageException extends Exception
{
	private String id;

	public MessageException(String id, String message)
	{
		super(message);
		this.id = id;
	}

	public MessageException(Throwable ex)
	{
		super(ex.getMessage());
		this.id = ex.getClass().getName();
	}

	public String getId()
	{
		return id;
	}
}

File: stasis-ios/Stasis/MessageException.m
//
//  SerializableException.m
//  Stasis
//
//  Created by Christian Fruth on 06.03.13.
//  Copyright (c) 2013 Boxx IT Solutions e.K. All rights reserved.
//

#import "MessageException.h"
#import "MessageExceptionSerializer.h"

@implementation MessageException

+ (instancetype)exceptionWithIdent:(NSString *)ident andMessage:(NSString *)message
{
	return [[MessageException alloc] initWithIdent:ident andMessage:message];
}

- (id)initWithIdent:(NSString *)ident andMessage:(NSString *)message
{
	return [super initWithName:ident reason:message userInfo:nil];
}

+ (NSString *)serializingAlias
{
	return @"de.boxxit.statis.MessageException";
}

+ (Class)defaultSerializer
{
	return [MessageExceptionSerializer class];
}

@end
